File: ATM_Project/MySQL/model/CreditCard.py
# /usr/bin/python
from ATM_Project.MySQL.model.Card import Card


class CreditCard(Card):
    INSERT = "INSERT INTO CreditCard(cardNum, owner, personID, phoneNum," \
             " password, deposit, creditLimit) VALUES (%s, %s, %s, %s, %s, %s, %s)"
    DELETE = "DELETE FROM CreditCard WHERE id = %s"
    UPDATE = "UPDATE CreditCard SET owner=%s, personID=%s, phoneNum=%s, password=%s," \
             "deposit=%s, creditLimit=%s WHERE id=%s"
    QUERY_ALL = "SELECT * FROM CreditCard"
    QUERY = "SELECT * FROM CreditCard WHERE id=%s"
    KIND = "CreditCard"

    def __init__(self):
        Card.__init__(self)
        self._creditLimit = 0

    def get_i_p(self):
        return (self.get_cardNum(), self.get_owner(), self.get_personID(), self.get_phoneNumber(),
                self.get_password(), self.get_deposit(), self.get_creditLimit())

    def get_u_p(self):
        return (self.get_owner(), self.get_personID(), self.get_phoneNumber(), self.get_password(),
                self.get_deposit(), self.get_creditLimit(), self.get_id())

    def to_string(self):
        Card.to_string(self)
        print("CreditCard limit:", self._creditLimit)

    def set_creditLimit(self, creditLimit):
        self._creditLimit = creditLimit

    def get_creditLimit(self):
        return self._creditLimit

File: java_atm/src/models/CreditCard.java
package models;

/**
 * Created by liuchuang on 25/02/16.
 */
public class CreditCard extends Card {
    private int creditLimit;

    public int getCreditLimit() {
        return creditLimit;
    }

    public void setCreditLimit(int creditLimit) {
        this.creditLimit = creditLimit;
    }

    public CreditCard() {
        super();
        this.creditLimit = 10000;
    }

    public String toString() {
        return "CreditCard{" +
                "id=" + this.getId() +
                ", cardNum='" + this.getCardNum() + '\'' +
                ", owner='" + this.getOwner() + '\'' +
                ", personID='" + this.getPersonID() + '\'' +
                ", phoneNum='" + this.getPhoneNum() + '\'' +
                ", password='" + this.getPassword() + '\'' +
                ", deposit=" + this.getDeposit() +
                ", creditLimit=" + this.getCreditLimit() +
                '}';
    }
}
